src/compiler/python_generator.cc

```diff
@@ -173,11 +116,62 @@ grpc::string ModuleAlias(const grpc::string& filename) {
   return module_name;
 }
 
-bool GetModuleAndMessagePath(const Descriptor* type,
-                             const ServiceDescriptor* service,
-                             grpc::string* out) {
+// Tucks all generator state in an anonymous namespace away from
+// PythonGrpcGenerator and the header file, mostly to encourage future changes
+// to not require updates to the grpcio-tools C++ code part. Assumes that it is
+// only ever used from a single thread.
+struct PrivateGenerator {
```

Comment:
It *could*, but the intent is that no code outside this file should ever need to access it and if anyone does then said person is almost certainly doing something wrong. Instead, such use is statically banned here.